File: lib/PCB.cpp
#include "../constants/processStates.cpp"
#include <vector>

using namespace std;

class PCB
{
private:
    /* data */
    int _state;
    PCB* _parent;
    vector<PCB*> _children; //  listi
    int _resources;
    int _priority;  //  extended


public:
    PCB(int priority);
    ~PCB();
    void test();
};

PCB::PCB(int priority)
{
    //  State = READY
    _state = READY; //1
    _priority = priority;
}

PCB::~PCB()
{
}

File: Project1_python/lib/PCB.py
import constants.priority as PRIORITY
import constants.states as STATE
import collections


class PCB:
    #   TODO: Add priority
    def __init__(self, parent, priority):
        self.parent = parent
        self.priority = priority
        self.children = collections.deque()
        self.state = STATE.READY
        self.resources = collections.deque()



    def __str__(self):
        return 'A {self._state} process'.format(self=self)